src/main/scala/Situation.scala

```diff
@@ -147,9 +147,8 @@ case class Situation(board: Board, color: Color):
   // TODO we depend on the correctness of epSQuare here
   private def genEnPassant(pawns: Bitboard): List[Move] =
     board.history.epSquare.fold(Nil)(ep =>
-      val pawnsCanEnPassant                       = pawns & ep.pawnAttacks(!color)
-      val ff: Bitboard => Option[(Pos, Bitboard)] = bb => bb.lsb.map((_, bb & (bb - 1L.bb)))
-      List.unfold(pawnsCanEnPassant)(ff).mapFilter(enpassant(_, ep))
+      val pawnsCanEnPassant = pawns & ep.pawnAttacks(!color)
+      pawnsCanEnPassant.fold(List[Move]())((moves, pos) => enpassant(pos, ep).fold(moves)(moves :+ _))
```

Comment:
Seems like this should be a foreach instead.

Reply:
The second fold is from `Option`. So, in case `enpassant` returns Some(move) we add that move to the results, otherwise just return the original results.

Reply:
Also the first fold is from [OpaqueBitboard](https://github.com/lichess-org/scalachess/blob/master/src/main/scala/bitboard/OpaqueBitboard.scala#L48).